caluma/extensions/visibilities.py

```diff
@@ -1,8 +1,24 @@
 from caluma.caluma_core.types import Node
 from caluma.caluma_core.visibilities import BaseVisibility, filter_queryset_for
+from caluma.caluma_form.schema import Document, Form, Option, Question
+from caluma.caluma_workflow.schema import Case, Flow, Task, Workflow, WorkItem
 
 
 class MySAGWVisibility(BaseVisibility):
     @filter_queryset_for(Node)
-    def filter_queryset_for_all(self, node, queryset, info):
-        return queryset
+    def filter_queryset(self, node, queryset, info):
+        user = info.context.user
+
+        if "sagw" in user.groups or "admin" in user.groups:
+            return queryset
+
+        if node in [Task, Flow, Workflow, Form, Option, Question]:
+            return queryset
+
+        if node in [WorkItem]:
```

Comment:
I suggest moving those specific cases into a specific `filter_queryset_for_xyz()` method. Let it fall back only in the generic case.

Reply:
As discussed in person.